model/src/main/kotlin/PackageReference.kt

```diff
@@ -72,16 +73,28 @@ data class PackageReference(
     fun collectDependencies(
         maxDepth: Int = -1,
         filterPredicate: (PackageReference) -> Boolean = { true }
-    ): SortedSet<Identifier> =
-        dependencies.fold(sortedSetOf()) { refs, ref ->
-            refs.also {
-                if (maxDepth != 0) {
-                    if (filterPredicate(ref)) it += ref.id
-                    it += ref.collectDependencies(maxDepth - 1, filterPredicate)
-                }
+    ): Set<Identifier> {
+        val result = mutableSetOf<Identifier>()
+
+        val queue = Stack<Pair<PackageReference, Int>>()
+        queue.push(Pair(this, 0))
+
+        while (queue.isNotEmpty()) {
+            val (pkg, level) = queue.pop()
+
+            if (filterPredicate(pkg)) {
```

Comment:
Nit: We tend to write conditionals as one-liners without braces if the condition is just a single statement.

Reply:
Done.